.github/workflows/react-compiler-compliance-v2.yml

```diff
@@ -0,0 +1,31 @@
+name: React Compiler Compliance Check v2
+
+on:
+  workflow_call:
+  pull_request:
+    types: [opened, synchronize]
+    branches-ignore: [staging, production]
+    paths: ["**.tsx"]
+
+concurrency:
+  group: ${{ github.ref == 'refs/heads/main' && format('{0}-{1}', github.ref, github.sha) || github.ref }}-react-compiler-compliance
```

Comment:
NAB but normally the unique part of the concurrency group is a suffix rather than a prefix